src/Razor/src/Microsoft.AspNetCore.Razor.LanguageServer/CodeActions/ComponentAccessibilityCodeActionProvider.cs

```diff
@@ -0,0 +1,219 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.AspNetCore.Razor.Language;
+using Microsoft.AspNetCore.Razor.Language.Legacy;
+using Microsoft.AspNetCore.Razor.Language.Syntax;
+using Microsoft.AspNetCore.Razor.LanguageServer.Common;
+using Microsoft.CodeAnalysis.Razor;
+using Microsoft.VisualStudio.Editor.Razor;
+using Newtonsoft.Json.Linq;
+using OmniSharp.Extensions.LanguageServer.Protocol.Models;
+
+namespace Microsoft.AspNetCore.Razor.LanguageServer.CodeActions
+{
+    internal class ComponentAccessibilityCodeActionProvider : RazorCodeActionProvider
+    {
+        private readonly TagHelperFactsService _tagHelperFactsService;
+        private readonly FilePathNormalizer _filePathNormalizer;
+
+        public ComponentAccessibilityCodeActionProvider(
+            TagHelperFactsService tagHelperFactsService,
+            FilePathNormalizer filePathNormalizer)
+        {
+            _tagHelperFactsService = tagHelperFactsService ?? throw new ArgumentNullException(nameof(tagHelperFactsService));
+            _filePathNormalizer = filePathNormalizer ?? throw new ArgumentNullException(nameof(filePathNormalizer));
+        }
+
+        override public Task<CommandOrCodeActionContainer> ProvideAsync(RazorCodeActionContext context, CancellationToken cancellationToken)
+        {
+            var commandOrCodeActions = new List<CommandOrCodeAction>();
+
+            // Locate cursor
+            var change = new SourceChange(context.Location.AbsoluteIndex, length: 0, newText: string.Empty);
+            var node = context.CodeDocument.GetSyntaxTree().Root.LocateOwner(change);
+            if (node is null)
+            {
+                return Task.FromResult<CommandOrCodeActionContainer>(null);
+            }
+
+            // Find start tag
+            var startTag = (MarkupStartTagSyntax)node.Ancestors().FirstOrDefault(n => n is MarkupStartTagSyntax);
+            if (startTag == null)
+            {
+                return Task.FromResult<CommandOrCodeActionContainer>(null);
+            }
+
+            // Ignore if has dots, as we only handle short tags
+            if (startTag.Name.Content.Contains("."))
+            {
+                return Task.FromResult<CommandOrCodeActionContainer>(null);
+            }
+
+            if (IsTagUnknown(startTag, context))
+            {
+                AddCreateComponentFromTag(context, startTag, commandOrCodeActions);
```

Comment:
Put this below  the "access" ones. Users  will more often then not  want  to add a "using"  and having  that at the  very top will make  it  easier to access.